core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -60,12 +62,19 @@ public void accept(String file) {
     }
   };
 
+  private final ExecutorService defaultDeleteExecutorService = Executors.newSingleThreadExecutor(runnable -> {
```

Comment:
I think the default should be the current behavior, which is to run immediately in the current thread. You can get that behavior by using [`MoreExecutors.directExecutor()`](https://guava.dev/releases/19.0/api/docs/com/google/common/util/concurrent/MoreExecutors.html#directExecutor()) as the default.

Reply:
If you use `directExecutor()` then this can be `static`.

Reply:
Using `MoreExecutors.newDirectExecutorService()` to keep it consistent with the expected type `java.util.concurrent.ExecutorService` to pass to `org.apache.iceberg.util.Tasks.Builder#executeWith`